src/Engine/Entity/Entity.cpp

```diff
@@ -255,3 +255,54 @@ unsigned int Entity::GetUniqueIdentifier() const {
 void Entity::SetUniqueIdentifier(unsigned int UID) {
     uniqueIdentifier = UID;
 }
+
+Component::SuperComponent* Entity::AddComponent(const std::type_info* componentType) {
```

Comment:
I have concerns with this. The entity can't possibly know _how_ components are to be created. It may very well know _where_ to do it, but it would be much more reasonable to create components outside the entity and pass created ones when it comes time to add them.

Reply:
Entities don't create components, though. They only delegate the responsibility to the respective manager.

Reply:
I don't claim that they do. This function assumes that creation of components requires no parameters (all methods that create components have empty parameter lists), and having that code here makes entity aware of this detail. It also prevents us from creating components in a fully working state unless every manager provides sensible defaults. My point is that it's more robust and flexible to pass along components that have already been created, thus moving this responsibility to the one who actually knows what details the component should be created with. In other words: we still create components via managers, just that this creation is done outside the entity where we know what parameters to pass to the manager.

Reply:
And where would that be?

Reply:
Wherever components are logically created (as in, we know what component we want added to an entity). On the top of my head: 1) when editor creates components, 2) when loading scene. In the latter case I would actually find it very clean if the managers were to have something like `CreateFromJson` or similar, but that's another issue. What do you think?